presto-main/src/main/java/com/facebook/presto/execution/buffer/BufferSize.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.buffer;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class BufferSize
+{
+    public static BufferSize emptySizes(String taskInstanceId, long token, boolean bufferComplete)
+    {
+        return new BufferSize(taskInstanceId, token, bufferComplete, ImmutableList.of());
+    }
+
+    private final String taskInstanceId;
+    private final long token;
+    private final boolean bufferComplete;
+    private final List<DataSize> pageSizes;
```

Comment:
Will change some other places (e.g., pendingRead, processRead, get() etc) from DataSize into long to avoid coverting long and DataSize back and forth